service/lxd_config.go

```diff
@@ -305,3 +305,37 @@ func (s LXDService) DefaultCephFSStoragePoolJoinConfig() api.ClusterMemberConfig
 		Value:  DefaultCephFSOSDPool,
 	}
 }
+
+// FindInterfaceForSubnet takes a subnet in CIDR notation (e.g., "192.168.1.0/24")
+// and returns the network interface that has an IP address in that subnet.
+func (s LXDService) FindInterfaceForSubnet(cidrSubnet string) (*net.Interface, error) {
+	_, targetNet, err := net.ParseCIDR(cidrSubnet)
+	if err != nil {
+		return nil, fmt.Errorf("Invalid subnet notation while looking for finding an interface: %v", err)
```

Comment:
`Invalid subnet notation: %w`

The suffix "while looking for an interface" can be appended by the caller of `FindInterfaceForSubnet` as it also applies to the other errors returned in this func.

Reply:
fixed